openai/openai.go

```diff
@@ -175,18 +176,29 @@ func New(opts ...Option) (*Client, error) {
 	if cfg.proxyURL != "" {
 		proxy, _ := url.Parse(cfg.proxyURL)
 		httpClient.Transport = &http.Transport{
-			Proxy: http.ProxyURL(proxy),
+			Proxy:               http.ProxyURL(proxy),
+			MaxIdleConns:        100,
+			MaxIdleConnsPerHost: 100,
+			TLSHandshakeTimeout: 5 * time.Second,
 		}
 	} else if cfg.socksURL != "" {
 		dialer, err := proxy.SOCKS5("tcp", cfg.socksURL, nil, proxy.Direct)
 		if err != nil {
 			return nil, fmt.Errorf("can't connect to the proxy: %s", err)
 		}
 		httpClient.Transport = &http.Transport{
-			Dial: dialer.Dial,
+			Dial:                dialer.Dial,
+			MaxIdleConns:        100,
+			MaxIdleConnsPerHost: 100,
+			TLSHandshakeTimeout: 5 * time.Second,
+		}
+	} else {
+		httpClient.Transport = &http.Transport{
+			MaxIdleConns:        100,
+			MaxIdleConnsPerHost: 100,
+			TLSHandshakeTimeout: 5 * time.Second,
 		}
 	}
-
 	c.HTTPClient = httpClient
 	instance.client = openai.NewClientWithConfig(c)
```

Comment:
Overall, the code patch seems to be well-structured and readable. Here are a few suggestions for improvement:

1. Error handling: There are a few places where errors are not being handled, such as when parsing the proxy URL and connecting to the proxy. It would be better to handle these errors and return an error message to the user.

2. Input validation: The function should validate the input values to ensure that they are valid URLs and that the user-provided options are correctly formatted.

3. Code duplication: There is some code duplication in the code patch, particularly in the creation of the HTTP transport. It would be better to create a separate function for this and reuse it across the codebase.

4. Security: The use of HTTP instead of HTTPS could be a security vulnerability. It would be better to use HTTPS whenever possible to ensure that data is encrypted in transit.

5. Performance: The MaxIdleConns and MaxIdleConnsPerHost values could be increased to improve performance, especially if the application is making many requests to the API.

Overall, the code patch seems to be of good quality, with only minor suggestions for improvement.